### PR TITLE
Add basic manual configuration

### PR DESCRIPTION
**Pull request by DCSBL**
Somehow I have issues with mDNS on my network and my Sync Box does not tell its there. This integration worked fine before but when I had to reinstall Home Assistant I was not able to set up my sync box.

With this workaround/feature I was able to configure my box. Maybe it is something you can use.

### CODE REVIEW

**Comment by DCSBL** [comment]
It works at least with my bridge, but there is no validation or error handling. What do you think?

**Review by mvdwetering — CHANGES_REQUESTED** [review]
Looks good.

I hoped to find and fix the actual issue for not discovering the box, but given that I have still not been able to reproduce the issue locally this might be the next best thing.

**Comment by mvdwetering** [comment]
> It works at least with my bridge, but there is no validation or error handling. What do you think?

I was wondering about this too, validation on valid IP and at least a length check on the ID would be nice. Bbut I looked around a bit in HA code but for for IP address there does not seem to be validation anywhere...

When inputting wrong data it does show a message with "Unknown error", but at least it does not explode and it keeps the previous values in the input dialog so after fixing it you can just continue, so not perfect, but very usable.

If it is easy to add some validation/errorhandling it would be nice, but not mandatory.

File: custom_components/huesyncbox/config_flow.py [hunk 1]
```diff
@@ -27,7 +29,29 @@ def __init__(self):
...
+                ),
+                errors=None,
+            )
+
+        entry_info = {
+            "host": user_input[CONF_IP_ADDRESS],
+            "port": 443,
+            "path": "/api",
+            "unique_id": user_input[CONF_UNIQUE_ID],
+            "name": "unknown_name",
```
[the remark this answers is not in the document]
Comment (DCSBL) [reply custom_components/huesyncbox/config_flow.py:52]: > I hoped to find and fix the actual issue for not discovering the box

Wait, is discovery not working on your box as well?

> we rename it to something like "manual entry" ?

I agree. Or just something like `Hue Sync Box`?

**Comment by DCSBL** [comment]
> HA code but for for IP address there does not seem to be validation anywhere

Validation in HA is 'Test if it works'. Because that validates if the IP is correct and working on one go. 
Maybe we can add some regex for the ID, or not make it required at all if that is possible with HTTPS.

File: custom_components/huesyncbox/config_flow.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvdwetering) [reply custom_components/huesyncbox/config_flow.py:52]: "Hue Sync Box" is actually better. Lets use that.

Discovery is working fine on my boxes which I why I can't really figure out what is wrong. For testing now I disabled the `zeroconf` part in the manifest.

File: custom_components/huesyncbox/strings.json [hunk 2]
```diff
@@ -27,7 +29,6 @@
       "cannot_connect": "Unable to connect to the device",
       "already_configured": "Device is already configured",
       "already_in_progress": "Config flow for device is already in progress.",
```
Comment (mvdwetering) [inline custom_components/huesyncbox/strings.json:31]: The Hassfest step in the Github action is complaining about the `strings.json`. I think it is the comma after this line. 
Just to be sure check the file with a validator.

**Review by mvdwetering — CHANGES_REQUESTED** [review]

**Comment by mvdwetering** [comment]
> > HA code but for for IP address there does not seem to be validation anywhere
> 
> Validation in HA is 'Test if it works'. Because that validates if the IP is correct and working on one go. Maybe we can add some regex for the ID, or not make it required at all if that is possible with HTTPS.

I ran into an errors because I forgot a dot in the IP, would have been nice to see the error before submitting, but I don't think HA does clientside checking (next to serverside) so it would not really help. Lets leave as is for now.

File: custom_components/huesyncbox/config_flow.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DCSBL) [reply custom_components/huesyncbox/config_flow.py]: https://github.com/mvdwetering/huesyncbox/pull/30/commits/c26c08ded9f8188673da2a69ae3d6b38f4d41a4f

File: custom_components/huesyncbox/translations/nb.json [hunk 3]
```diff
@@ -3,9 +3,11 @@
     "flow_title": "{name}",
     "step": {
       "user": {
-        "title": "Koble til enheten",
+        "title": "Enter device information",
+        "description": "Details can be found in the Hue Sync Box App under 'Network and Device information'.",
         "data": {
-          "host": "Vert"
+          "ip_address": "IP Address",
+          "unique_id": "Identifier (eg. C42996000000)"
```
Comment (DCSBL) [inline custom_components/huesyncbox/translations/nb.json:10]: I'm not that good in Norwegian, @hwikene can you take a look?

File: custom_components/huesyncbox/strings.json (hunk 2 above)
Comment (DCSBL) [reply custom_components/huesyncbox/strings.json]: b3133a2

**Review by mvdwetering — APPROVED** [review]

**Comment by mvdwetering** [comment]
I merged it, any language updates can be done with separate PRs

**Comment by mvdwetering** [comment]
FYI I finally made a release with these changes. Sorry it took so long, but I was working on some other things that I wanted to include, but I ran into some issues there so I finally decided to not delay further and release this feature.

Again thanks for the addition!

**Comment by DCSBL** [comment]
No worries! I’ve learned a lot of this integration to make the HomeWizard Energy integration. Great that I was able to do something back :)

**Pull request merged**